### DISCUSSION TITLE
google oauth error

### DISCUSSION THREAD

**Discussion by Tristan-2021** [supabase/supabase#3658] [Questions] [answered]
google oauth error 400 redirect_uri_mismatch android, I don't know what I'm doing wrong as I set everything up from your supabase and google, I hope someone can help me. I have also thought that it does not yet have support to create an account with google

**Comment by awalias** [comment, marked as answer]
hey @Tristan-2021 !

have you put your redirect url (`https://<project-ref>.supabase.co/auth/v1/callback`) into the google dashboard? more info here: https://supabase.io/docs/guides/auth/auth-google#find-your-callback-url

are you using the hosted version of supabase, or self-hosted?

**Reply by Tristan-2021** [reply]
Thanks for the prompt response. I already solved it, it turns out that when configuring OAuth 2.0 client ID I have done it for android and now I did it for the web, there the supabase link already works and I can start session with google from the device. what I'm wondering is why when configuring with android it doesn't work?

**Comment by thomergil** [comment]
The only thing that fixed it for me -- even though I had correctly configured the Redirect URI -- was to create a new OAuth Client ID with exactly the same specs.

**Reply by wzulfikar** [reply]
uh.. this actually works. spent 2 hrs checking all possible mistakes, and the fix is actually recreating the client

**Comment by gndps** [comment]
It seems supabase deliberately try to hide the complete documentation for the local setup, it should be as straighforward as the hosted instance.
I got it working with these steps: https://gemini.google.com/share/0d2862e4f9f6
Note that using `locahost` vs `127.0.0.1` in the strings can make a difference

**Reply by aantti** [reply]
Just added:
- https://supabase.com/docs/guides/self-hosting/self-hosted-oauth
- https://supabase.com/docs/guides/self-hosting/self-hosted-phone-mfa
